# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: 23d50b5617de
# Full Hash: 23d50b5617def9d4152023fd98210c0a9723e3c0
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1774704
# File Overlap Count: 1
# Description:
#   Bug 1774704 - part 4-3: Make `HTMLEditor::MaybeSplitAncestorsForInsertWithTransaction` stop computing editing host with current selection r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152968
# ==============================================================================

diff -r 310cd44cff95 -r 23d50b5617de editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Thu Aug 04 02:02:46 2022 +0000
+++ b/editor/libeditor/HTMLEditor.h	Thu Aug 04 02:12:03 2022 +0000
@@ -1300,14 +1300,15 @@
    * @param aTag                        The name of element to be inserted
    *                                    after calling this method.
    * @param aStartOfDeepestRightNode    The start point of deepest right node.
-   *                                    This point must be descendant of
-   *                                    active editing host.
+   *                                    This point must be in aEditingHost.
+   * @param aEditingHost                The editing host.
    * @return                            When succeeded, SplitPoint() returns
    *                                    the point to insert the element.
    */
   [[nodiscard]] MOZ_CAN_RUN_SCRIPT SplitNodeResult
   MaybeSplitAncestorsForInsertWithTransaction(
-      nsAtom& aTag, const EditorDOMPoint& aStartOfDeepestRightNode);
+      nsAtom& aTag, const EditorDOMPoint& aStartOfDeepestRightNode,
+      const Element& aEditingHost);
 
   /**
    * InsertElementWithSplittingAncestorsWithTransaction() is a wrapper of
